masschallenge/impact-api
# AC-5993 Improve handling of clearance-style permission checks on impact-api

## [AC-5993](https://masschallenge.atlassian.net/browse/AC-5993)

This ticket arose because impact-api had a local, buggy check for clearance level (it failed if the user's permission was higher *or* lower than requested, not just if it was lower). 

Per the "ideal solution" described in the ticket, the changes here (across three repos) relocate the Clearance manager code to django-accelerator, and uses that for the impact-api `global_operations_manager_check`.

(Also see [accelerate PR](https://github.com/masschallenge/accelerate/pull/1894) and [django-accelerator PR](https://github.com/masschallenge/django-accelerator/pull/96))

**To test:**
* Log into as a user with `CLEARANCE_LEVEL_EXEC_MD` clearance
* Load a view that uses the `global_operations_manager_check` (e.g. [CloneCriteriaView](https://github.com/masschallenge/impact-api/blob/AC-5993/web/impact/impact/v1/views/clone_criteria_view.py)
* Confirm that you are allowed to see the page (i.e. you don't get a 403)
